apps/dashr-lastodash/app.R

```diff
@@ -0,0 +1,379 @@
+      las_data$V <- data.frame(x = contentV) %>% 
+        separate(col = x, into = c("label", "x"), sep = "\\.", extra = "merge") %>% 
+        separate(col = x, into = c("value", "description"), sep = ":\\s+", extra = "merge")
+    }
+    # store section containing well identification
+    if (startsWith(section, "W")) {
+      linesW <- str_split(section, "\r\n")
+      contentW <- linesW[[1]][-c(1, length(linesW[[1]]))]
+      if ((length(contentW) >=  2) & startsWith(contentW, "#")) {
+        contentW <- contentW[-c(1, 2)]
+      }
+      las_data$W <- data.frame(x = contentW) %>%
+        separate(col = x, into = c("mnemonic", "x"), sep = "\\.", extra = "merge") %>% 
+        separate(col = x, into = c("unit", "x"), sep = "\\s+", extra = "merge") %>% 
+        separate(col = x, into = c("value", "description"), sep = ":\\s+", extra = "merge")
+    }
+    # store section containing curve information
+    if (startsWith(section, "C")) {
+      linesC <- str_split(section, "\r\n")
+      contentC <- linesC[[1]][-c(1, length(linesC[[1]]))]
+      if ((length(contentC) >=  2) & startsWith(contentC, "#")) {
+        contentC <- contentC[-c(1, 2)]
+      }
+      las_data$C <- data.frame(x = contentC) %>%
+        separate(col = x, into = c("mnemonic", "x"), sep = "\\.", extra = "merge") %>%
+        separate(col = x, into = c("unit", "x"), sep = "\\s+", extra = "merge") %>%
+        separate(col = x, into = c("api code", "curve description"), sep = ":\\s+", extra = "merge")
+    }
+    # section contains ASCII log data
+    if (startsWith(section, "A")) {
+      contentA <- str_sub(section, 2, -2)
+      las_data$A <- read.table(text = contentA, header = TRUE, sep = "")
+    }
+  }
+  return (las_data)
+}
+
+las_data <- convert_las(las_path)
+
+table_header <- list (
+  "mnemonic" = list(name = "mnemonic", id = "mnemonic", width = "10%"),
+  "description" = list(name = "description", id = "description", width = "40%"),
+  "unit" = list(name = "unit", id = "unit", width = "10%"),
+  "value" = list(name = "value", id = "value", width = "40%")
+)
+
+####################################################################################################
+# DEFINE FUNCTIONS FOR APP LAYOUT
+
+generate_frontpage <- function() {
+  desc <- str_trim(las_data$V[[3]][1], "left")
+  return(
+    list(
+      htmlDiv(
+        id = "las-header",
+        children = list(
+          htmlImg(id = "las-logo", src = "assets/logo.png"),
+          htmlDiv(
+            id = "las-header-text",
+            children = list(
+              htmlH1("LAS Report"),
+              htmlDiv(
+                id = "las-file-info",
+                children = list(
+                  htmlSpan(id = "las-filename", children = las_file),
+                  htmlSpan(glue(" ({desc})"))
+                )
+              )
+            )
+          )
+        )
+      )
+    )
+  )
+}
+
+generate_table <- function() {
+  return(
+    dashTable::dashDataTable(
```

Comment:
The namespace for `dashTable` should get loaded when you `library(dashTable)`, so we can :hocho: `dashTable::` here.